src/server/worker/workerrequests.js

```
@@ -248,14 +268,37 @@ function WorkerRequests(mainLogger, gmeConfig, webgmeUrl) {
                 if (typeof socketId === 'string') {
                     logger.debug('socketId provided for plugin execution - notifications available.');
                     pluginManager.notificationHandlers = [function (data, callback) {
+                        if (data.notification.type && data.notification.type ===
+                            STORAGE_CONSTANTS.PLUGIN_NOTIFICATION_TYPE.INITIATED) {
+                            data.executionId = context.executionId;
+                            data.pluginSocketId = storage.webSocket.socket.id;
+                        }
                         data.originalSocketId = socketId;
                         storage.sendNotification(data, callback);
                     }];
                 }
 
                 pluginManager.projectAccess = res.access;
 
-                pluginManager.executePlugin(pluginName, context.pluginConfig, pluginContext, finish);
+                // pluginManager.executePlugin(pluginName, context.pluginConfig, pluginContext, finish);
+
+                pluginManager.initializePlugin(pluginName)
+                    .then(function (plugin_) {
+                        plugin = plugin_;
+                        return pluginManager.configurePlugin(plugin, context.pluginConfig, pluginContext);
+                    })
+                    .then(function () {
+                        pluginManager.runPluginMain(plugin, finish);
+                    })
+                    .catch(function (err) {
+                        var pluginResult = pluginManager.getPluginErrorResult(plugin.id,
+                            pluginName,
+                            'Exception was raised, err: ' + err.stack,
+                            plugin && plugin.projectId);
+
+                        pluginManager.logger.error(err.stack);
+                        finish(err.message, pluginResult);
+                    });
```

This can fail silently (add .done())